grid_helper.js

```diff
@@ -352,28 +353,42 @@ function buildView(divId, initialLayout, initialContent=[], numCols=12, gridRowH
         if (window.gridHelperOptions) {
           this.kioskMode = window.gridHelperOptions.kioskMode;
           this.tableaux = window.gridHelperOptions.tableaux || ['Welcome', 'Comic', 'Gallery'];
-          this.defaultTableauName = window.gridHelperOptions.defaultTableauName || this.tableaux[0] || 'welcome';
+          this.defaultTableauName = window.gridHelperOptions.defaultTableauName || this.tableaux[0] || 'Welcome';
+          this.locationHash = window.location.hash;
         }
         else {
           this.tableaux = ['Welcome', 'Comic', 'Gallery'];
           this.defaultTableauName = this.tableaux[0];
+          this.locationHash = window.location.hash;
         }
       },
       mounted: function() {
         const that = this;
         window.addEventListener('resize', that.onResize);
-
-        if (initialContent && initialLayout) {
-          this.sdContent = initialContent;
-          this.layout = initialLayout;
-          this.applyContentAndFixLayout();
+        window.onhashchange = that.locationHashChanged;
+        //window.addEventListener('hashChange', that.locationHashChanged);
+        if (that.locationHash && that.locationHash.length !== 0) {
+          // User given a specific tableau to navigate to
+          var tableauName = that.locationHash.replace(/#/g, '');
+          that.loadTableau(tableauName);
         }
-        else {
-          loadTableau(`tableaux/${this.defaultTableauName}.yaml`, function(layout, sdContent) {
-            that.layout = layout;
-            that.sdContent = sdContent;
-            that.applyContentAndFixLayout();
-          });
+        else if (that.locationHash == '') {
+          // User did not provide any tableau to load, use default or debug content
+          if (initialContent && initialLayout) {
+            this.sdContent = initialContent;
+            this.layout = initialLayout;
+            this.applyContentAndFixLayout();
+            this.locationHash = '#Debug';
```

Comment:
I recommend setting the hash to the empty string `''` instead of `#Debug`. That way, the unadorned URL (e.g., `http://127.0.0.1:8080)`) will refer to the 'Debug' content (assuming that you set `debugLayout = null;` below (see next comment).